Code/Framework/AzCore/Tests/Time/TimeTests.cpp

```diff
@@ -77,4 +76,191 @@ namespace UnitTest
+        AZ::TimeUs diff = GetDiff(delta, baselineDelta);
+        EXPECT_LT(diff, AZ::TimeUs{ 10 });
+
+        // the delta should be the same as GetSimulationTickDeltaTimeUs and near GetRealTickDeltaTimeUs
+        const AZ::TimeUs simDeltaTime = m_timeSystem->GetSimulationTickDeltaTimeUs();
+        EXPECT_EQ(delta, simDeltaTime);
+
+        const AZ::TimeUs realDeltaTime = m_timeSystem->GetRealTickDeltaTimeUs();
+        diff = GetDiff(delta, realDeltaTime);
+        EXPECT_LT(diff, AZ::TimeUs{ 10 });
+    }
+
+    TEST_F(TimeSystemTests, SimulationAndRealTickDeltaTimesWithTimeScale)
+    {
+        // advance the tick delta to get a clean base.
+        m_timeSystem->AdvanceTickDeltaTimes();
+        const AZ::TimeUs baselineStart = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond());
+
+        // slow down 'time';
+        m_timeSystem->SetSimulationTickScale(0.5f);
+
+        // sleep for a bit to advance time.
+        AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(2));
+
+        // advance the tick delta.
+        const AZ::TimeUs delta = m_timeSystem->AdvanceTickDeltaTimes();
+        const AZ::TimeUs baselineDelta = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond()) - baselineStart;
+        const AZ::TimeUs halfBaselineDelta = (baselineDelta / AZ::TimeUs{ 2 });
+
+        // the delta should be half the baselineDelta
+        AZ::TimeUs diff = GetDiff(delta, halfBaselineDelta);
+        EXPECT_LT(diff, AZ::TimeUs{ 10 });
+
+        // the delta should be the same as GetSimulationTickDeltaTimeUs
+        const AZ::TimeUs simDeltaTime = m_timeSystem->GetSimulationTickDeltaTimeUs();
+        EXPECT_EQ(delta, simDeltaTime);
+
+        // the delta should be near half the GetRealTickDeltaTimeUs
+        const AZ::TimeUs realDeltaTime = m_timeSystem->GetRealTickDeltaTimeUs();
+        const AZ::TimeUs halfRealDeltaTime = (realDeltaTime / AZ::TimeUs{ 2 });
+        diff = GetDiff(delta, halfRealDeltaTime);
+        EXPECT_LT(diff, AZ::TimeUs{ 10 });
+
+        // reset time scale
+        m_timeSystem->SetSimulationTickScale(1.0f);
+    }
+
+    TEST_F(TimeSystemTests, SimulationTickDeltaOverride)
+    {
+        // advance the tick delta to get a clean base.
+        m_timeSystem->AdvanceTickDeltaTimes();
+        const AZ::TimeUs baselineStart = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond());
+
+        // set the tick delta override to 1 millisecond
+        m_timeSystem->SetSimulationTickDeltaOverride(AZ::TimeMs{ 1 });
+
+        // sleep for a bit to advance time.
+        AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(2));
+
+        // advance the tick delta.
+        const AZ::TimeUs delta = m_timeSystem->AdvanceTickDeltaTimes();
+        const AZ::TimeUs baselineDelta = static_cast<AZ::TimeUs>(AZStd::GetTimeNowMicroSecond()) - baselineStart;
+
+        // the delta should be 1 millisecond
+        EXPECT_EQ(delta, AZ::TimeUs{ 1000 });
```

Comment:
suggestion: To match the override with the expected value, maybe the override value could be put in a variable and use later as the expected delta. Also, use a less trivial number than 1.

````
const auto overrideDelta = AZ::TimeUs{ 3546 };
m_timeSystem->SetSimulationTickDeltaOverride(overrideDelta);
...
EXPECT_EQ(delta, overrideDelta);
````